### PR TITLE
cleanup: Remove remaining use of the deprecated google.protobuf.Struct

### PR DESCRIPTION
**Pull request by yanavlasov**
Remove remaining use of the deprecated google.protobuf.Struct in configuration in the source code.
Doc cleanup is in a separate PR.

Part of fixing #5558

Risk Level: Low
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: source/extensions/filters/network/thrift_proxy/config.cc [hunk 1]
```diff
@@ -162,7 +162,7 @@ void ConfigImpl::processFilter(
   ENVOY_LOG(debug, "    thrift filter #{}", filter_factories_.size());
   ENVOY_LOG(debug, "      name: {}", string_name);
   ENVOY_LOG(debug, "    config: {}",
-            MessageUtil::getJsonStringFromMessage(proto_config.config(), true));
+            MessageUtil::getJsonStringFromMessage(proto_config.typed_config(), true));
```
Comment (htuch) [inline source/extensions/filters/network/thrift_proxy/config.cc]: What does this look like interactively? Can you paste a Gist?

Comment (yanavlasov) [reply source/extensions/filters/network/thrift_proxy/config.cc]: Looks fine. See https://pastebin.com/KxzeMyMJ

**Comment by mattklein123** [comment]
@yanavlasov qq: do we still have test coverage of the deprecated paths?

/wait-any

**Comment by yanavlasov** [comment]
> @yanavlasov qq: do we still have test coverage of the deprecated paths?
> 
> /wait-any
It is quite limited. There are some tests in https://github.com/envoyproxy/envoy/blob/master/test/common/router/config_impl_test.cc#L6440

I may be able to make tests using Struct elsewhere. When I looked at doing this for integration tests it ended being a can of worms.

**Comment by yanavlasov** [comment]
> Thanks, this is really great and helps with the v3 migration. Do you know if there is an easy way to ban future use of plain `config()` with `check_format`? I think it might be hard, because it's a fairly common string, but maybe worth a quick look.

It is doable. We'd need to whitelist about 20 files that use .config() or ->config()
I'm unsure about any future use though.

**Comment by htuch** [comment]
As discussed IRL, let's skip the check_format given that `config()` is on it way out.

**Comment by yanavlasov** [comment]
> Thanks, this is really great and helps with the v3 migration. Do you know if there is an easy way to ban future use of plain `config()` with `check_format`? I think it might be hard, because it's a fairly common string, but maybe worth a quick look.

We can wait to see if we'll be getting PRs that use untyped config. If we do I'll add check_format checks.

**Comment by htuch** [comment]
@yanavlasov can you merge master post API boost? Is there anything blocking us merging this after that?

**Comment by yanavlasov** [comment]
> @yanavlasov can you merge master post API boost? Is there anything blocking us merging this after that?

Done. MacOS CI failure seems unrelated, looks like memory calculations in the stats integration tests are off by a few bytes.

**Comment by htuch** [comment]
@jmarantz FYI on the stats integration test and `envoy-macos` failure; I saw the same in other PRs today.

**Pull request merged**